File: kotlin-orders/kotlin-orders-dto/src/main/java/com/mvv/bank/orders/rest/entities/enums.kt
package com.mvv.bank.orders.rest.entities


enum class OrderType {
    MARKET_ORDER,
    LIMIT_ORDER,
    STOP_ORDER,
}


enum class Side {
    CLIENT,
    BANK_MARKET,
}


enum class BuySellType {
    BUY,
    SELL,
}


enum class DailyExecutionType {
    DAY_ONLY,
    GTC,
}


enum class OrderState {
    UNKNOWN,
    TO_BE_PLACED,
    PLACED,
    EXECUTED,
    EXPIRED,
    CANCELED,
}

File: scala-orders/src/main/scala/com/mvv/bank/orders/rest/entities/enums.scala
package com.mvv.bank.orders.rest.entities

enum OrderType derives CanEqual :
  case MARKET_ORDER
     , LIMIT_ORDER
     , STOP_ORDER


enum Side derives CanEqual :
  case CLIENT, BANK_MARKET


enum BuySellType derives CanEqual :
  case BUY, SELL


enum DailyExecutionType derives CanEqual :
  case DAY_ONLY
     , GTC


enum OrderState derives CanEqual :
  case UNKNOWN
     , TO_BE_PLACED
     , PLACED
     , EXECUTED
     , EXPIRED
     , CANCELED
